modules/c++/mem/include/mem/BufferView.h

```diff
@@ -50,21 +51,52 @@ struct BufferView
     T* data;
     size_t size;
 
-    std::vector<BufferView> split(size_t n)
-    {
-        const size_t newSize = size / n;
-        const size_t lastSize = size - (n - 1) * newSize;
-        T *head = data;
-
-        std::vector<BufferView> buffers(n); 
-        for (size_t ii = 0, last_ii = n - 1; ii < n; ++ii, head += newSize)
-        {
-            buffers[ii] = BufferView(head,
-                                     (ii == last_ii) ? lastSize : newSize);
-        }             
-        return buffers;
-    }                 
+    // returns a new bufferView that "takes" sectionSize T from the bufferView
+    // For instance, if the current bufferView had 100 ints, and you called 
+    // section with sectionSize = 10, section() would return a new BufferView
+    // of size 10 and the original BufferView would be left with 90 ints.
+    // throws std::string if there is not enough space 
+    mem::BufferView<T> section(size_t sectionSize);
+
+    // returns a vector of as many n-sized BufferViews as possible
+    // the last BufferView will have the remainder (size % n) elements if
+    // size % n != 0
+    std::vector<BufferView> split(size_t n);
+
 };
+
+
+template <typename T>
+mem::BufferView<T> mem::BufferView<T>::section(size_t sectionSize)
+{
+    if(size < sectionSize)
+    {
+        throw std::string("section tried to take more space than available");
```

Comment:
Throw an `except::Exception(Ctxt("Blah blah"))`.  Expand it to tell you how much space you had and how much you asked for.  Include `<sys/Conf.h>` and `except/Exception.h` to get this stuff.